pkg/scheduler/scheduler_test.go

```diff
@@ -0,0 +1,273 @@
+/*
+Copyright (c) Microsoft Corporation.
+Licensed under the MIT license.
+*/
+
+package scheduler
+
+import (
+	"context"
+	"log"
+	"os"
+	"strconv"
+	"testing"
+
+	"github.com/google/go-cmp/cmp"
+	"github.com/google/go-cmp/cmp/cmpopts"
+	"k8s.io/apimachinery/pkg/api/errors"
+	metav1 "k8s.io/apimachinery/pkg/apis/meta/v1"
+	"k8s.io/apimachinery/pkg/types"
+	"k8s.io/client-go/kubernetes/scheme"
+	"sigs.k8s.io/controller-runtime/pkg/client/fake"
+
+	fleetv1beta1 "go.goms.io/fleet/apis/placement/v1beta1"
+)
+
+const (
+	crpName = "test-crp"
+
+	bindingName    = "test-binding"
+	altBindingName = "another-test-binding"
+
+	policySnapshotName        = "test-policy-snapshot"
+	altPolicySnapshotName     = "another-test-policy-snapshot"
+	anotherPolicySnapshotName = "yet-another-test-policy-snapshot"
+)
+
+var (
+	ignoreObjectMetaResourceVersionField = cmpopts.IgnoreFields(metav1.ObjectMeta{}, "ResourceVersion")
+	ignoreTypeMetaAPIVersionKindFields   = cmpopts.IgnoreFields(metav1.TypeMeta{}, "APIVersion", "Kind")
+)
+
+// TestMain sets up the test environment.
+func TestMain(m *testing.M) {
+	// Add custom APIs to the runtime scheme.
+	if err := fleetv1beta1.AddToScheme(scheme.Scheme); err != nil {
+		log.Fatalf("failed to add custom APIs to the runtime scheme: %v", err)
+	}
+
+	os.Exit(m.Run())
+}
+
+// TestCleanUpAllBindingsFor tests the cleanUpAllBindingsFor method.
+func TestCleanUpAllBindingsFor(t *testing.T) {
+	now := metav1.Now()
+	crp := &fleetv1beta1.ClusterResourcePlacement{
+		ObjectMeta: metav1.ObjectMeta{
+			Name:              crpName,
+			DeletionTimestamp: &now,
+			Finalizers:        []string{fleetv1beta1.SchedulerCRPCleanupFinalizer},
+		},
+	}
+
+	bindings := []*fleetv1beta1.ClusterResourceBinding{
+		{
+			ObjectMeta: metav1.ObjectMeta{
+				Name: bindingName,
+				Labels: map[string]string{
+					fleetv1beta1.CRPTrackingLabel: crpName,
+				},
+			},
+		},
+		{
+			ObjectMeta: metav1.ObjectMeta{
+				Name: altBindingName,
+				Labels: map[string]string{
+					fleetv1beta1.CRPTrackingLabel: crpName,
+				},
+			},
+		},
+	}
+
+	fakeClient := fake.NewClientBuilder().
+		WithScheme(scheme.Scheme).
+		WithObjects(crp, bindings[0], bindings[1]).
+		Build()
+	// Construct scheduler manually instead of using NewScheduler() to avoid mocking the controller
+	// manager.
+	s := &Scheduler{
+		client:         fakeClient,
+		uncachedReader: fakeClient,
+	}
+
+	ctx := context.Background()
+	if err := s.cleanUpAllBindingsFor(ctx, crp); err != nil {
+		t.Errorf("cleanUpAllBindingsFor() = %v, want no error", err)
```

Comment:
use fatalf, as there is no need to validate outputs

Reply:
Done.